### DISCUSSION TITLE
Some feature requests

### DISCUSSION THREAD

**Discussion by noconsulate** [supabase/supabase#1131] [Feature Requests]
Dear devs,

Thanks for the great product! Here are a few ideas I've written down...

1. **Short IDs** - One thing I love about Firebase, and I also found this feature in Parse, is the short primary IDs it gives you. I was able to use the short ID as a URL parameter, making sharing an instance of my game with other players very easy. Incremental IDs let people join a game they're not invited to and UUIDs are way too long to put in a URL and you can't write them down. I've looked at ways to implement this myself and it's not easy to do. I would think it wouldn't be terribly hard for Supabase to implement because you'd just have to some process keeping track of IDs for every supbase-project/table. I feel like short primary IDs is one of those Firebase features that a lot of people might take for granted but doesn't get talked about very much.
2. **Get rid of parameter hints in editor** - I personally find the editor very difficult to use having to constantly press escape to see the lines below my cursor. I'd rather just have a PSQL shell! Fortunately you make my database accessible for me to connect to with another editor. 
3. **Retain state between views** - In the dashboard when I go from SQL -> Tables Editor -> SQL I lose my open editor and have to make an additional click. At least it keeps my open tabs. It's not a big deal but it's annoying and there's no reason to show me that default screen with all the built-in scripts every time.

Thanks again for what you do.

**Comment by kiwicopple** [comment]
These are all great suggestions. I'll make sure the team are aware of them all. 

> Short IDs 

Worth checking out this extension! https://github.com/turbo/pg-shortkey. It's just SQL so you can run it in your Supabase Dashboard.

**Reply by noconsulate** [reply]
Ah, very nice. I will check that out.

**Reply by cloudorbush** [reply]
The following error appears when trying to change the id type to `shortkey`:

`identity column type must be smallint, integer, or bigint`

**Comment by noconsulate** [comment]
I've got a few more ideas. Assuming someone looks at old posts I'll just keep putting my ideas in replies to this thread so that I don't pollute the board.

1. Delete all users. I create a lot of junk users when building my app. It's probably a bad practice on my part, but I do it anyway. It'd be nice to delete all my users with one (or two) clicks. Perhaps there's something I can type in the editor to do this? I'd have to research the other projects you're built upon. 
2. This goes with idea # 1: It'd be nice to turn OFF the magic link. Especially when I'm developing and inputting junk users I'd hate to accidentally put in a real email address and have someone get spammed by Supabase. Even in production it'd be nice to have to option to turn off this feature.
3. Alter the order of columns in a table. I realize there's no built in way to easily to do this in Postgres but I imagine you could rig it up to just make a new table and copy the data over. Or if this is too intensive make it so that the table editor is actually looking at a view rather than the table directly. However you achieve it, it'd be real nice and I'm sure your users will be looking for that feature. I have just set up the pg-shortkey that @kiwicopple suggested and now I have my ID column all the way at the end which means I have to manually make a new table just to get the thing looking right, otherwise the order of columns will annoy me for as long as this project lasts.

**Reply by steve-chavez** [reply]
Related: 3 has been brought up before on https://github.com/supabase/supabase/discussions/622.
